cms/0011-block-data-structure.md

```diff
@@ -1,140 +1,129 @@
-[
-  {
-    "settings": {
-      "setting1": true,
-      "setting2": "blue"
-    }, 
-    "content": {
-      "udi": "umb://element/fffba547615b4e9ab4ab2a7674845bc9",
-      "title": "Hello"
-    }
-  },
-  {
-    "settings": {
-      "setting1": false,
-      "setting2": "red"
-    }, 
-    "content": {
-      "udi": "umb://element/e7dba547615b4e9ab4ab2a7674845bc9",
-      "title": "World"
-    }
-  }
-]
-```
-
-### Layout
-
-Alongside the list of blocks stored, the editor will also store a Layout. The Layout will reference each content item in the list of blocks by it's UDI. The Layout object in the json will be key/value pairs where the `key` is the Property Editor alias. This is required because some editors will not store a simple one dimensional array layout structure such as Nested Content, other more complicated editors like the grid will store references to the blocks with it's own layout structure. By defining each layout with the alias of the Property Editor it means developers can in theory swap the underlying property editor without losing data and while keeping the layout preserved by the previous editor.
-
-#### Example of full JSON
+#### Simple example of JSON
 
 ```json
 {
   "layout": {
-    "Umbraco.NestedContent": ["umb://element/fffba547615b4e9ab4ab2a7674845bc9", "umb://element/e7dba547615b4e9ab4ab2a7674845bc9"]
-  },
-  "blocks": [
-    {
-      "settings": {
-        "setting1": true,
-        "setting2": "blue"
-      }, 
-      "content": {
+    "Umbraco.NestedContent": [
+      {
         "udi": "umb://element/fffba547615b4e9ab4ab2a7674845bc9",
-        "title": "Hello"
-      }
-    },
-    {
-      "settings": {
-        "setting1": false,
-        "setting2": "red"
+        "settings": {}
       }, 
-      "content": {
+      {
         "udi": "umb://element/e7dba547615b4e9ab4ab2a7674845bc9",
-        "title": "World"
+        "settings": {}
       }
+    ]
+  },
+  "data": [
+    {
+      "udi": "umb://element/fffba547615b4e9ab4ab2a7674845bc9",
+      "title": "Hello"
+    },
+    {
+      "udi": "umb://element/e7dba547615b4e9ab4ab2a7674845bc9",
+      "title": "World",
+      "text": "The content of a textarea"
     }
   ]
 }
 
 ```
 
+### Layout
+The Layout will contain an object with a structure specific to the Property Editor, named by the Property Editor Alias.
+This is required because some editors will not store a simple one-dimensional array such as Nested Content, but rather multi-dimensional structures like Grid Layouts. By defining each layout with the alias of the Property Editor it means developers can in theory swap the underlying Property Editor without losing data and while keeping the layout preserved by the previous editor.
+
+The structure of a given layout object is defined by the Property Editor and can therefore vary depending on the Property Editor.
+
+Common for all is that the object refferencing content items would implement the `IBlockElement` interface. This interface has two properties:
```

Comment:
Spelling "refferencing".

This is actually not entirely true any longer. The `settings` can be an element type (IPublishedElement) but doesn't necessarily need to be. A property editor can in theory choose to have settings that are non-editable. For NL's example, a new version of the grid may not define visually editable settings (i.e. grid columns) and instead these values would be implicitly set by the editor (i.e. drag/drop column widths). In those cases, the settings wont be an element type, they will just be data defined by the editor.  In other cases, like potentially the Block Editor, the property editor may allow the administrator to choose an Element Type to be used as the settings element. In which case the settings would be an `IPublishedElement`.

So really, the strongly typed `IBlockElement` interface needs to be redefined:

```
public interface IBlockElement<TSettings>
{
    Udi Udi { get; }
    TSettings Settings { get; }
}
```

with not explicit requirement that TSettings is `IPublishedElement` and the type `TSettings` could be defined by the property editor developer.